### PR TITLE
Named Function References Type Annotation

### PR DESCRIPTION
**Pull request by MTimmerije**
# Pull Request

## Description

Implements type annotations for the named function reference type.

It closes a subtask of #58 

## Changes

1. Implemented the type annotation
2. Added it to the AST

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

**Review by yuanzexiong — APPROVED** [review]
Decent stuff great work!

File: src/expressions/NamedFunctionRef.ts [hunk 1]
```diff
@@ -22,7 +23,8 @@ class NamedFunctionRef extends Expression {
 
 	constructor(
 		functionReference: { localName: string; namespaceURI: string | null; prefix: string },
-		arity: number
+		arity: number,
+		type?: SequenceType
```
Comment (FBroy) [inline src/expressions/NamedFunctionRef.ts]: I don't think this needs to be optional?

Comment (MTimmerije) [reply src/expressions/NamedFunctionRef.ts]: fair point

**Review by maanpalle — APPROVED** [review]

File: src/expressions/NamedFunctionRef.ts (hunk 1 above)
Comment (MTimmerije) [reply src/expressions/NamedFunctionRef.ts]: Fixed this.

**Review by FBroy — APPROVED** [review]
Looks good! Good job

**Pull request merged**